src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,219 @@
+    private User first;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        first = new User();
+    }
+
+    @Test
+    void register_validAge_ok() {
+        first.setAge(19);
+        first.setLogin("first1");
+        first.setPassword("passw1");
+        User actual = registrationService.register(first);
+        User expected = storageDao.get(first.getLogin());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_minValidAge_ok() throws RegistrationException {
+        first.setAge(18);
+        first.setLogin("second2");
+        first.setPassword("passw2");
+        User actual = registrationService.register(first);
+        User expected = storageDao.get(first.getLogin());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_maxValidAge_ok() throws RegistrationException {
+        first.setAge(110);
+        first.setLogin("second3");
+        first.setPassword("passw3");
+        User actual = registrationService.register(first);
+        User expected = storageDao.get(first.getLogin());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_lessThanMinValidAge_notOk() {
+        first.setAge(6);
+        first.setLogin("firsgddt1");
+        first.setPassword("passfew1");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(first);
+        });
+    }
+
+    @Test
+    void register_greaterThanMaxAge_notOk() {
+        first.setAge(116);
+        first.setLogin("firsttt");
+        first.setPassword("passwww");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(first);
+        });
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        first.setAge(-11);
+        first.setLogin("firsrcttt");
+        first.setPassword("67sswww");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(first);
+        });
+    }
+
+    @Test
+    void register_nullUserValue_notOk() {
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_ageIsNull_notOk() {
+        first.setLogin("seco76nd2");
+        first.setPassword("pasgggsw2");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(first);
+        });
+    }
+
+    @Test
+    void register_validPassword_ok() {
+        first.setAge(19);
+        first.setLogin("firserfdt1");
+        first.setPassword("passdfsw1");
+        User actual;
+        try {
+            actual = registrationService.register(first);
+        } catch (RegistrationException e) {
+            return;
+        }
+        User expected = storageDao.get(first.getLogin());
```

Comment:
we almost never use try/catch in tests. If exception is thrown and we haven't explicitly declared that we expect it, test will fail